server/lib/matchmaking/matchmaking-api.ts

```diff
@@ -106,6 +108,20 @@ export class MatchmakingApi {
     })
     const { clientId, preferences, identifiers } = body
 
+    const currentMapPool = await getCurrentMapPool(preferences.matchmakingType)
+    if (!currentMapPool) {
+      throw new MatchmakingServiceError(
+        MatchmakingServiceErrorCode.InvalidMapPool,
+        "Map pool doesn't exist",
+      )
+    }
+    if (preferences.mapSelections.length > currentMapPool.maxVetoCount) {
```

Comment:
This error isn't a good addition. If the player previously had vetoes set and the next map pool decreases the count, they can end up being over the count, and now you're just stopping them from matchmaking. Just truncate their vetoes to the proper size.

Alongside that, you should first remove any vetoes that are not maps in the current pool, *before* doing that truncation.